### DISCUSSION TITLE
'MOS lock in' protection reset?

### DISCUSSION THREAD

**Discussion by ithody** [KG-Development/SmartBMSUtility#4] [Q&A]
Hi,
I have been experiencing some issues using your app (which looks amazing, BTW) in conjunction with my installed smart bluetooth BMS, which I would really appreciate some help on. I am simply testing my newly built 13S5P battery pack with the BMS installed, with no loading for any discharge testing. So right now, just interested in charging and balancing functions.

The app connects to the device with no issues and is able to read/write configuration data from/to it. (I've updated the start balance voltage to 3800mv to test balancing without fully charging.) Everything looks good but an error (MOS lock in) is highlighted (turns from green to red) when I access the "Protection" status screen. I'm thinking this may relate to the MOSFET circuitry being switch off, for some reason, which would likely cause the charging and/or discharge not to function. Please can you clarify what this relates to, what would likely have caused it, and what I should do to reset it. (I've tried disconnecting/re-connecting the BMS balance wire loom to try to reset the BMS to eliminate the error without success.)

The app has separate buttons on the Overview screen to invoke "Charging" and "Discharging" functions. Is this to support BMS boards that have discrete outputs for charging and discharging? On my BMS, both charge and discharge use the same (C-) output port on the PCB. When I try to invoke the "Charging" function, the BMS disconnects the C- output and charging is disabled. However, if I invoke the "Discharging" function (with no load attached) the battery looks to be charging ok.  Even though the "MOS lock in" error is still evident. Is this expected behaviour? Should I simply ignore the "Charging" option in the GUI? Or am I missing something?

Thanks for any help or advice you're able to offer.

Regards, Ian

**Comment by NeariX67** [comment]
Hey,

you are completely right, the "MOS lock in" means that either or both the charging and discharging ports have been disabled. This seems weird, when no other error is beeing thrown, as that protection feature should only activate when an over- or undervoltage is detected. Are your charge and discharge button on the overview on (green)?

**Reply by ithody** [reply]
Thanks for the quick response.
Yes, both the charge and discharge buttons are displayed as green. When either is invoked it stays green but indicates 'busy' (or 'executing') symbol.

**Reply by NeariX67** [reply]
Yeah, thats sadly a bug that i didn't knew to fix when i coded that part. I hope i can fix this until v1.1.2.
Can you please completely close the app and connect back to the bms? Are the two buttons still green? This might be caused by the behaviour of the bug described earlier.

**Reply by ithody** [reply]
OK, closed the app, removed from iPhone memory, and restarted it. Overview screen shows both charging and discharging buttons as green. I have not presses either of them as the battery still appears to be charging OK without invoking either. Protection screen still shows same error.

**Reply by NeariX67** [reply]
Hmm. Thats weird. Its definitely not related to my app, but to the bms itself. Yet I have no idea whats causing this.

**Reply by ithody** [reply]
Incidentally, maybe a dumb question but ... are the "Charging" and "Discharging" buttons intended to switch those functions on and off from the GUI? Or just to highlight if those functions are available, rather than disabled, on the BMS?

**Reply by NeariX67** [reply]
Yes, the two buttons indeed have funtion. They will block or unblock the charge / discharge functions of the bms. So you can disable charging for example and still be able to draw power from your battery

**Reply by ithody** [reply]
Ah, that would explain why charging stops when I press the "Charging" button.  (Silly me!) Presumably, the button should turn red when pressed and the function disabled. Rather than indicating 'busy'. The bug you mentioned earlier?

So, as things stand, all appears to working correctly on the charging side of things, even with the error on the Protection screen. Remains to be seen whether this will prohibit discharge when I have a load added. Whatever the case, this is down to the BMS state itself and not caused by the app.

Thanks again for your help and time on this.  Great work on the app!

**Reply by NeariX67** [reply]
The button *should* indeed turn red when charging is disabled, but you guessed right - the bug is causing this misbehaviour. I hope i get to update the app until this weekend

**Reply by ithody** [reply]
One last question ... should the app display differently when the BMS goes into balance mode for individual cells? I'm not seeing any indication that balancing is taking place now the battery has charged cells past the 3800mV threshold I've set in the app. Or am I misunderstanding something?  
Thanks.

**Reply by NeariX67** [reply]
Yeah, there are some lightning symbols in the cellvoltages segment (Overview page). But the BMS has a really annoying design flaw: Setting "Balancing only whilst charging" in the configuration page will balance ONLY when you are *obviously* charging. Disabling that will only balance them when they are NOT charging... Totally stupid, but it's nothing we have control of.

**Reply by ithody** [reply]
Unless the BMS has some form of active (rather than passive) balancing between cells, which would make sense to offer during discharge as well as charging.  But I didn't think that was the case from what I've read on these boards. Otherwise, as you say, the option is completely superfluous.

**Reply by NeariX67** [reply]
> Yes, both the charge and discharge buttons are displayed as green. When either is invoked it stays green but indicates 'busy' (or 'executing') symbol.

It appears that I already fixed this issue in v1.1.0. Does this issue still persist? I was unable to get my charging/discharging button stuck in loading
